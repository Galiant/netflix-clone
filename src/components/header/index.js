/*eslint-disable react/prop-types*/
import React from 'react';
import { Link as ReactRouterLink } from 'react-router-dom';
import {
  Background,
  Container,
  Logo,
  ButtonLink,
  Feature,
  Text,
  FeatureText,
} from './styles/header';

export default function Header({ bg = true, children, ...restProps }) {
  return bg ? <Background {...restProps}>{children}</Background> : children;
}

Header.Feature = function HeaderFeature({ children, ...restProps }) {
  return <Feature {...restProps}>{children}</Feature>;
};

Header.FeatureText = function HeaderFeatureText({ children, ...restProps }) {
  return <FeatureText {...restProps}>{children}</FeatureText>;
};

Header.Text = function HeaderText({ children, ...restProps }) {
  return <Text {...restProps}>{children}</Text>;
};

Header.Frame = function HeaderFrame({ children, ...restProps }) {
  return <Container {...restProps}>{children}</Container>;
};

Header.ButtonLink = function HeaderButtonLink({ children, ...restProps }) {
  return <ButtonLink {...restProps}>{children}</ButtonLink>;
};

Header.Logo = function HeaderLogo({ to, ...restProps }) {
  return (
    <ReactRouterLink to={to}>
      <Logo {...restProps} />
    </ReactRouterLink>
  );
};
