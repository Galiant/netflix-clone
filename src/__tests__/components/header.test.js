import React from 'react';
import userEvent from '@testing-library/user-event';
import { render } from '@testing-library/react';
import { Header } from '../../components';

jest.mock('react-router-dom');

describe('Header', () => {
  it('renders basic Header without a background', () => {
    const { container, getByText, queryByTestId } = render(
      <Header bg={false}>
        <Header.Frame>
          <Header.Logo alt='Netflix' src='/logo.svg' />
          <Header.TextLink active='true'>Hello I am a link!</Header.TextLink>
        </Header.Frame>
      </Header>
    );

    expect(getByText('Hello I am a link!')).toBeTruthy();
    expect(queryByTestId('header-bg')).toBeFalsy();
    expect(container.firstChild).toMatchSnapshot();
  });

  it('renders full Header with a background', () => {
    const { container, getByText, getByTestId } = render(
      <Header src='joker1' dontShowOnSmallViewPort>
        <Header.Frame>
          <Header.Group>
            <Header.Logo src='/images/logo.svg' alt='Netflix' />
            <Header.TextLink active={false} onClick={() => {}}>
              Series
            </Header.TextLink>
            <Header.TextLink active={false} onClick={() => {}}>
              Films
            </Header.TextLink>
          </Header.Group>
          <Header.Group>
            <Header.Search
              searchTerm='Joker'
              setSearchTerm={() => {}}
            ></Header.Search>
            <Header.Profile>
              <Header.Picture src='/images/karl.png' />
              <Header.Dropdown>
                <Header.Group>
                  <Header.Picture src='/images/karl.png' />
                  <Header.TextLink>Antonijo</Header.TextLink>
                </Header.Group>
                <Header.Group>
                  <Header.TextLink onClick={() => {}}>Sign out</Header.TextLink>
                </Header.Group>
              </Header.Dropdown>
            </Header.Profile>
          </Header.Group>
        </Header.Frame>
        <Header.Feature>
          <Header.FeatureText>Watch Joker Now</Header.FeatureText>
          <Header.Text>Forever alone in a crowd...</Header.Text>
          <Header.PlayButton>Play</Header.PlayButton>
        </Header.Feature>
      </Header>
    );
    expect(getByTestId('search-input')).toBeTruthy();
    expect(getByTestId('search-input').value).toBe('Joker');
    userEvent.type(getByTestId('search-input'), 'Simpsons');
    userEvent.click(getByTestId('search-click'));
    expect(getByText('Series')).toBeTruthy();
    expect(getByText('Films')).toBeTruthy();
    expect(getByText('Antonijo')).toBeTruthy();
    expect(getByText('Watch Joker Now')).toBeTruthy();
    expect(getByText('Sign out')).toBeTruthy();
    expect(getByText('Forever alone in a crowd...')).toBeTruthy();
    expect(getByText('Play')).toBeTruthy();
    expect(container.firstChild).toMatchSnapshot();
  });
});
