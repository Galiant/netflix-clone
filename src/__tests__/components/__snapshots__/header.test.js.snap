// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Header renders basic Header without a background 1`] = `
<div
  class="header__Container-t8lkk2-3 dsukoR"
>
  <p
    class="header__Link-t8lkk2-5 fYkiTA"
  >
    Hello I am a link!
  </p>
</div>
`;

exports[`Header renders full Header with a background 1`] = `
<div
  class="header__Background-t8lkk2-0 cwUAtN"
  data-testid="header-bg"
  src="joker1"
>
  <div
    class="header__Container-t8lkk2-3 dsukoR"
  >
    <div
      class="header__Group-t8lkk2-1 hNdlqq"
    >
      <p
        class="header__Link-t8lkk2-5 jvjwEc"
      >
        Series
      </p>
      <p
        class="header__Link-t8lkk2-5 jvjwEc"
      >
        Films
      </p>
    </div>
    <div
      class="header__Group-t8lkk2-1 hNdlqq"
    >
      <div
        class="header__Search-t8lkk2-7 iiBTmq"
      >
        <button
          class="header__SearchIcon-t8lkk2-8 dMRHCr"
          data-testid="search-click"
        >
          <img
            alt="Search"
            src="/images/icons/search.png"
          />
        </button>
        <input
          class="header__SearchInput-t8lkk2-9 crdRlb"
          data-testid="search-input"
          placeholder="Search films and series"
          value="Joker"
        />
      </div>
      <div
        class="header__Profile-t8lkk2-11 jfbVpf"
      >
        <button
          class="header__Picture-t8lkk2-4 dHQAYc"
          src="/images/users//images/karl.png.png"
        />
        <div
          class="header__Dropdown-t8lkk2-6 jIQDsg"
        >
          <div
            class="header__Group-t8lkk2-1 hNdlqq"
          >
            <button
              class="header__Picture-t8lkk2-4 dHQAYc"
              src="/images/users//images/karl.png.png"
            />
            <p
              class="header__Link-t8lkk2-5 jvjwEc"
            >
              Antonijo
            </p>
          </div>
          <div
            class="header__Group-t8lkk2-1 hNdlqq"
          >
            <p
              class="header__Link-t8lkk2-5 jvjwEc"
            >
              Sign out
            </p>
          </div>
        </div>
      </div>
    </div>
  </div>
  <div
    class="header__Container-t8lkk2-3 header__Feature-t8lkk2-14 dsukoR dYkPzg"
  >
    <h2
      class="header__FeatureText-t8lkk2-16 bzSdnR"
    >
      Watch Joker Now
    </h2>
    <p
      class="header__Text-t8lkk2-15 daqyUi"
    >
      Forever alone in a crowd...
    </p>
    <button
      class="header__PlayButton-t8lkk2-10 jnWiig"
    >
      Play
    </button>
  </div>
</div>
`;
